test-suite/shared/src/test/scala/org/scalajs/testsuite/javalib/util/MapTest.scala

```diff
@@ -1217,12 +1228,48 @@ trait MapTest {
     assertTrue(mp.remove("ONE", "one"))
     assertFalse(mp.containsKey("ONE"))
 
+    if (factory.allowsNullKeys) {
+      mp.put(null, "one")
+      assertFalse(mp.remove(null, "not exist"))
+      assertTrue(mp.containsKey(null))
+      assertTrue(mp.remove(null, "one"))
+      assertFalse(mp.containsKey(null))
+    } else {
+      assertThrows(classOf[NullPointerException], mp.remove(null, "old value"))
+    }
+
     if (factory.allowsNullValues) {
       mp.put("nullable", null)
       assertFalse(mp.remove("nullable", "value"))
       assertTrue(mp.containsKey("nullable"))
       assertTrue(mp.remove("nullable", null))
       assertFalse(mp.containsKey("nullable"))
+    } else {
+      // Currently, ConcurrentHashMap is the only null-rejecting Map implemented in Scala.js.
+      // JavaDoc does not describe that CHM#remove(key, null) should throw NPE, and actually
+      // they do not throw NPE both on JDK 8 & 11. So we do not assert NPE thrown.
+      // However, it should be asserted that mp#(key, null) should not remove.
+      // https://bugs.java.com/bugdatabase/view_bug.do?bug_id=6272521
```

Comment:
This is not the right comment style. Mutli line comments should use: `/*` (see the coding style guidelines).

I also don't necessarily agree with your interpretation that the spec is ambiguous (or at least this is how I read the text). IMO if it *doesn't* say it throws, it shouldn't. So maybe, the comment can simply be removed.

Reply:
Updated https://github.com/scala-js/scala-js/pull/4367/commits/cd4ce0f68f72fcfce8b0bb21100bba50c97f6dbe